client/ayon_core/pipeline/create/creator_plugins.py

```diff
@@ -551,6 +564,16 @@ def get_instance_attr_defs(self):
 
         return self.instance_attr_defs
 
+    def get_attr_defs_for_instance(self, instance):
+        """Get attribute definitions for an instance.
+
+        Args:
+            instance (CreatedInstance): Instance for which to get
+                attribute definitions.
+
+        """
+        return self.get_instance_attr_defs()
```

Comment:
So - _how_ dynamic is this? Like, from the `CreatedInstance` to what data in it are we allowed to respond to make the returned attribute definitions unique? E.g. technically we can now differentiate attribute defs based on "other attributes" on the instance - however I wonder whether that's intended 'feature'?

- Can we rely on doing something different based on e.g. folder path or task? If so, what happens if someone without resets confirms a target folder path change from within the UI? Does it refresh?
- Can we rely on doing something differently based on e.g. `families` on the instance if it has it?
- Can we rely on doing something based off of other attributes on the instance from other plug-ins, I suppose not? Since it'd only update on reset?

Reply:
> Can we rely on doing something differently based on e.g. families on the instance if it has it?

Yes, but if I may request, please do not use `families` for it if possible. We can discuss each individual use-case (not here).

Why: In case of USD, you probably should be adding `families` based on the definitions, instead of showing definitions based on families? Meaning, you can add `"isUsdInstance": True` on instance data to know if is related to USD or not (please take this as very very basic example).

> Can we rely on doing something different based on e.g. folder path or task? If so, what happens if someone without resets confirms a target folder path change from within the UI? Does it refresh?

> Can we rely on doing something based off of other attributes on the instance from other plug-ins, I suppose not? Since it'd only update on reset?

Both question have same answer. Right now it collects definitions only when instance is created or collected (on create and on reset). For "real update" we need callbacks. Future PR mentioned in description, which I think might require change of the methods or create context api.

Reply:
> Why: In case of USD, you probably should be adding families based on the definitions, instead of showing definitions based on families? Meaning, you can add "isUsdInstance": True on instance data to know if is related to USD or not (please take this as very very basic example).

I see, but preferably we work towards something similar to `families` (like standardizes like that, alsmost like "tags") but for the instance's features instead instead of resulting in TONS and TONS of `isUsdInstance=True` and `isFbxInstance=True`. Stepping away from `pyblish` terminology like `families`, yes! But just doing for example `targetProductTypes = {"fbx", "usd"}` or whatever name we come up seems like something we can standardize way better. Or maybe better yet, have a dedicated attribute for that on the instance. E.g. being able to do something like: ```python
if instance.supports("usd"):
    # bla
``` 

Likely overlaps tons with @antirotor work with OpenAssetIO but that's a can of worms I'm afraid to touch. :)

---

> Both question have same answer. Right now it collects definitions only when instance is created or collected (on create and on reset). For "real time update" we need callbacks. Future PR mentioned in description, which I think might require change of the methods or create context api.

👍

Reply:
Because I don't know the context, so I can't react. Only from what you wrote I have 10 following questions that will raise another questions. Maybe with full context I would agree, but at this moment I don't.

It is possible to do anything, we should just discuss what would be the ideal approach, to make it clear from data flow.

Like I've said (wrote) we can discuss that out of this PR, as it is not directly related, and it requires more minds.